src/server/pfs/fuse/server.go

```diff
@@ -1000,8 +1080,20 @@ func mountedState(m *MountStateMachine) StateFn {
 			// TODO: handle remount case (switching mode):
 			// if mounted ro and switching to rw, just upgrade
 			// if mounted rw and switching to ro, upload changes, then switch the mount type
-		} else {
+		case "unmount":
 			return unmountingState
+		case "commit":
+			return committingState
+		default:
+			m.responses <- Response{
+				Repo:       m.MountKey.Repo,
+				Branch:     m.MountKey.Branch,
+				Commit:     m.MountKey.Commit,
+				Name:       m.Name,
+				MountState: m.MountState,
+				Error:      fmt.Errorf("bad request: unsupported/unknown action %s while mounted", req.Action),
+			}
+			return unmountedState
```

Comment:
There's a distinction between an error with the request ("bad request") and the mount itself being in an error state (couldn't communicate with pachyderm, or some other internal error). Kinda like the difference between HTTP 4xx and 5xx. This is a "bad request" case and it doesn't make sense to error the whole mount just because a user sent a bad request to it.

Reply:
Makes sense!